ironfish-rust/src/note.rs

```diff
@@ -196,25 +195,69 @@ impl<'a> Note {
     /// actually read the contents.
     pub fn encrypt(&self, shared_secret: &[u8; 32]) -> [u8; ENCRYPTED_NOTE_SIZE + aead::MAC_SIZE] {
         let mut bytes_to_encrypt = [0; ENCRYPTED_NOTE_SIZE];
-        bytes_to_encrypt[..11].copy_from_slice(&self.owner.diversifier.0[..]);
-        bytes_to_encrypt[11..43].clone_from_slice(self.randomness.to_repr().as_ref());
+        bytes_to_encrypt[..32].clone_from_slice(self.randomness.to_repr().as_ref());
 
-        LittleEndian::write_u64_into(&[self.value], &mut bytes_to_encrypt[43..51]);
-        bytes_to_encrypt[51..].copy_from_slice(&self.memo.0[..]);
+        LittleEndian::write_u64_into(&[self.value], &mut bytes_to_encrypt[32..40]);
+        bytes_to_encrypt[40..].copy_from_slice(&self.memo.0[..]);
         let mut encrypted_bytes = [0; ENCRYPTED_NOTE_SIZE + aead::MAC_SIZE];
         aead::encrypt(shared_secret, &bytes_to_encrypt, &mut encrypted_bytes);
 
         encrypted_bytes
     }
 
+    /// Computes the note commitment, returning the full point.
+    fn cm_full_point(&self) -> jubjub::SubgroupPoint {
```

Comment:
What do you think about renaming this? I think it's helpful to use a more clear naming scheme
```suggestion
    fn commitment_full_point(&self) -> jubjub::SubgroupPoint {
```